src/components/transactions/SingleTx/index.tsx

```diff
@@ -1,59 +1,53 @@
-import { CircularProgress } from '@mui/material'
+import { Box, CircularProgress } from '@mui/material'
 import ErrorMessage from '@/components/tx/ErrorMessage'
-import useChainId from '@/hooks/useChainId'
-import { useRouter } from 'next/router'
 import useSafeInfo from '@/hooks/useSafeInfo'
-import useAsync from '@/hooks/useAsync'
-import type { DateLabel, Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
-import { getTransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
-import { sameAddress } from '@/utils/addresses'
+import type { Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
 import type { ReactElement } from 'react'
-import { makeDateLabelFromTx, makeTxFromDetails } from '@/utils/transactions'
+import { makeTxFromDetails } from '@/utils/transactions'
 import { TxListGrid } from '@/components/transactions/TxList'
-import TxDateLabel from '@/components/transactions/TxDateLabel'
 import ExpandableTransactionItem from '@/components/transactions/TxListItem/ExpandableTransactionItem'
+import { useTxDetails } from '@/hooks/useTxDetails'
+import { isMultisigDetailedExecutionInfo } from '@/utils/transaction-guards'
+import { GroupLabelTypography, useFutureNonceLabel } from '../GroupLabel'
+import { useRouter } from 'next/router'
 
 const SingleTxGrid = ({ txDetails }: { txDetails: TransactionDetails }): ReactElement => {
   const tx: Transaction = makeTxFromDetails(txDetails)
-  const dateLabel: DateLabel = makeDateLabelFromTx(tx)
 
   return (
     <TxListGrid>
-      <TxDateLabel item={dateLabel} />
       <ExpandableTransactionItem item={tx} txDetails={txDetails} testId="single-tx" />
     </TxListGrid>
   )
 }
 
 const SingleTx = () => {
-  const chainId = useChainId()
   const router = useRouter()
   const { id } = router.query
-  const transactionId = Array.isArray(id) ? id[0] : id
-  const { safe, safeAddress } = useSafeInfo()
-
-  const [txDetails, txDetailsError] = useAsync<TransactionDetails>(
-    () => {
-      if (!transactionId) return
-      return getTransactionDetails(chainId, transactionId)
-    },
-    [transactionId, safe.txQueuedTag, safe.txHistoryTag],
-    false,
-  )
-  const isCurrentSafeTx = sameAddress(txDetails?.safeAddress, safeAddress)
+  const [txDetails, error] = useTxDetails(Array.isArray(id) ? id[0] : id)
+  const futureNonceLabel = useFutureNonceLabel()
 
-  const error = !transactionId
-    ? new Error("Couldn't retrieve the transaction details. Please review the URL.")
-    : !isCurrentSafeTx
-    ? new Error(`Transaction with id ${transactionId} not found in this Safe`)
-    : txDetailsError
+  const { safe } = useSafeInfo()
+
+  let nonceWarning: string | undefined = undefined
+  const executionInfo = txDetails?.detailedExecutionInfo
+  if (isMultisigDetailedExecutionInfo(executionInfo)) {
+    if (executionInfo.nonce > safe.nonce) {
```

Comment:
This is much simpler (I like it) than the [DateLabel logic.](https://github.com/safe-global/web-core/blob/af182a29562c00a50b8e9eb83aa0e1bc6f4bab44/src/components/transactions/GroupLabel/index.tsx#L25-L32)

Why do we need that complicated logic there and not here?